OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -82,11 +82,12 @@ void RescanForTargets(Actor self)
 
 			// If close enough, we don't care about other actors.
 			var target = self.World.FindActorsInCircle(self.CenterPosition, WormInfo.IgnoreNoiseAttackRange)
-				.FirstOrDefault(x => attackTrait.HasAnyValidWeapons(Target.FromActor(x)));
-			if (target != null)
+				.Select(t => Target.FromActor(t))
+				.FirstOrDefault(t => attackTrait.HasAnyValidWeapons(t));
+
+			if (target.Type == TargetType.Actor)
 			{
-				self.CancelActivity();
-				attackTrait.ResolveOrder(self, new Order("Attack", target, true) { TargetActor = target });
+				attackTrait.AttackTarget(target, false, true, false);
```

Comment:
If those checks failed (which isn't possible with the default d2k worm definitions) then not cancelling and continuing to move would be the more correct option.